### PR TITLE
WIP: ElasticSearch version 7.2 test adaptation

### PR DESCRIPTION
**Pull request by LanKhuat**
Follow up for: https://github.com/apache/james-project/pull/288/commits

### CODE REVIEW

File: backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java [hunk 1]
```diff
@@ -104,30 +96,16 @@ void indexMessageShouldThrowWhenJsonIsNull() {
     }
     
     @Test
-    @Disabled("JAMES-3492")
     void updateMessages() {
         String content = "{\"message\": \"trying out Elasticsearch\",\"field\":\"Should be unchanged\"}";
 
         testee.index(DOCUMENT_ID, content, useDocumentId(DOCUMENT_ID)).block();
-        elasticSearch.awaitForElasticSearch();
+        awaitForElasticSearch(QueryBuilders.matchAllQuery(),1L);
```
Comment (Arsnael) [inline backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java:103]: space missing between your arguments

File: backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java [hunk 2]
```diff
@@ -165,22 +143,14 @@ void deleteByQueryShouldWorkOnSingleMessage() {
         RoutingKey routingKey = useDocumentId(documentId);
 
         testee.index(documentId, content, routingKey).block();
-        elasticSearch.awaitForElasticSearch();
-        
+        awaitForElasticSearch(QueryBuilders.matchAllQuery(),1L);
```
Comment (Arsnael) [inline backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java:146]: idem

File: backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java [hunk 3]
```diff
@@ -165,22 +143,14 @@ void deleteByQueryShouldWorkOnSingleMessage() {
...
-        
-        CALMLY_AWAIT.atMost(Duration.TEN_SECONDS)
-            .until(() -> client.search(
-                    new SearchRequest(INDEX_NAME.getValue())
-                        .source(new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())),
-                    RequestOptions.DEFAULT)
-                .block()
-                .getHits().getTotalHits().value == 0);
+
+        awaitForElasticSearch(QueryBuilders.matchAllQuery(),0L);
```
Comment (Arsnael) [inline backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java:150]: idem

File: backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java [hunk 4]
```diff
@@ -196,68 +166,45 @@ void deleteByQueryShouldWorkWhenMultipleMessages() {
         String content3 = "{\"message\": \"trying out Elasticsearch 3\", \"property\":\"2\"}";
         
         testee.index(documentId3, content3, ROUTING).block();
-        elasticSearch.awaitForElasticSearch();
+        awaitForElasticSearch(QueryBuilders.matchAllQuery(),3L);
```
Comment (Arsnael) [inline backends-common/elasticsearch-v7/src/test/java/org/apache/james/backends/es/v7/ElasticSearchIndexerTest.java:169]: idem

**Review by chibenwa — COMMENTED** [review]
We will have to understand if we did introduce a breaking change removing that default _mapping stuff but it looks OK so far...

**Comment by chibenwa** [comment]
It is :green_apple: .

Any opposition about merging it as is and continuing this work?

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
I'm not really opposed to merge this except that the ES migration has still not been merged itself

**Comment by Arsnael** [comment]
Has been merged, but still need a bit more work

**Pull request closed without merging**